### PR TITLE
[wasm] Add tests for build with SIMD disabled

### PR DESCRIPTION
**Pull request by radekdoulik**
Add BuildWithoutSIMD_AOT and PublishWithoutSIMD_AOT tests

Also try to enable SIMD tests on node

### CODE REVIEW

**Comment by radekdoulik** [comment]
> should we also explicitly disable simd for some of the benchmarks now?

I plan to remove simd flavors and replace them with nosimd flavors once the things stabilize.

File: src/mono/wasm/Wasm.Build.Tests/WasmSIMDTests.cs [hunk 1]
```diff
@@ -48,13 +48,41 @@ public void BuildWithSIMD_NoAOT_ShouldNotRelink(BuildArgs buildArgs, RunHost hos
...
+                                    new BuildProjectOptions(
+                                        InitProject: () => File.WriteAllText(Path.Combine(_projectDir!, "Program.cs"), s_simdProgramText),
+                                        Publish: false,
+                                        DotnetWasmFromRuntimePack: true));
+
+            if (!_buildContext.TryGetBuildFor(buildArgs, out _))
+            {
+                // Check if this is not a cached build
+                Assert.Contains("Compiling native assets with excc", output);
+            }
```
Comment (radical) [inline src/mono/wasm/Wasm.Build.Tests/WasmSIMDTests.cs:69]: Ilona fixed `BuildProject` to always return the build output. So, this assert can be unconditional. And also the string has a typo - `excc` -> `emcc`.

File: src/mono/wasm/Wasm.Build.Tests/WasmSIMDTests.cs [hunk 2]
```diff
@@ -31,11 +38,10 @@ public void BuildWithSIMD_NoAOT_ShouldNotRelink(BuildArgs buildArgs, RunHost hos
...
                                         DotnetWasmFromRuntimePack: true));
 
-            if (!_buildContext.TryGetBuildFor(buildArgs, out _))
-            {
-                // Check if this is not a cached build
-                Assert.Contains("Compiling native assets with excc", output);
-            }
+            _buildContext.TryGetBuildFor(buildArgs, out _);
+
+            // Check if this is not a cached build
```
Comment (radical) [inline src/mono/wasm/Wasm.Build.Tests/WasmSIMDTests.cs:43]:
```suggestion
            // Confirm that we didn't relink
```

**Review by radical — APPROVED** [review]
Other than the comment, LGTM 👍

**Pull request merged**